app/services/ai_backend/gemini.rb

```diff
@@ -0,0 +1,112 @@
+class AIBackend::Gemini < AIBackend
+  include Tools
+  class ::Gemini::Errors::ConfigurationError < ::Gemini::Errors::GeminiError; end
+
+  # Rails system tests don't seem to allow mocking because the server and the
+  # test are in separate processes.
+  #
+  # In regular tests, mock this method or the TestClients::Gemini class to do
+  # what you want instead.
+  def self.client
+    if Rails.env.test?
+      ::TestClient::Gemini
+    else
+      ::Gemini
+    end
+  end
+
+  def initialize(user, assistant, conversation = nil, message = nil)
+    super(user, assistant, conversation, message)
+    begin
+      raise configuration_error if assistant.api_service.requires_token? && assistant.api_service.effective_token.blank?
+      Rails.logger.info "Connecting to Gemini API server at #{assistant.api_service.url} with access token of length #{assistant.api_service.effective_token.to_s.length}"
+      @client = self.class.client.new(credentials: {service: "generative-language-api",
+                                                    api_key: assistant.api_service.effective_token,
+                                                    version: "v1beta"},
+                                      options: { model: assistant.language_model.api_name,
+                                      server_sent_events: true })
+    rescue ::Faraday::UnauthorizedError, ::Faraday::BadRequestError => e
+      raise configuration_error
+    end
+  end
+
+  def client_method_name
+    :stream_generate_content
+  end
+
+  def configuration_error
+    ::Gemini::Errors::ConfigurationError
+  end
+
+  def set_client_config(config)
+    super(config)
+
+    @client_config =  {
+      contents: config[:messages],
+      system_instruction: ( system_message(config[:instructions]) if @assistant.language_model.supports_system_message?)
+    }.compact
+  end
+
+  def get_oneoff_message(instructions, messages, params = {})
+    response = @client.generate_content( { system_instruction: system_message(instructions), contents: { role: "user", parts: { text: messages.first }}})
+    response.dig("candidates",0,"content","parts",0,"text")
+  end
```

Comment:
Related to my other comment, this method accepts params but it isn't being passed through. That will make this get_oneoff_message() more versatile, such as supporting JSON response type, and it'll make it better match the other models.